submitit/snapshot.py

```diff
@@ -0,0 +1,61 @@
+import itertools
+import os
+import shutil
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Sequence
+
+
+def run_cmd(str_args, **kwargs):
+    return subprocess.check_output(str_args, **kwargs).decode("utf-8").strip()
+
+
+class SnapshotManager:
```

Comment:
as @gwenzek mentioned, maybe RsyncSnapshotManager (or actually jusct RsyncSnapshot?). That clarifies the need for Rsync + lets the door open to other variants